packages/mui-base/src/Portal/Portal.tsx

```diff
@@ -33,8 +34,10 @@ const Portal = React.forwardRef(function Portal(
 ) {
   const { children, container, disablePortal = false } = props;
   const [mountNode, setMountNode] = React.useState<ReturnType<typeof getContainer>>(null);
-  // @ts-expect-error TODO upstream fix
-  const handleRef = useForkRef(React.isValidElement(children) ? children.ref : null, forwardedRef);
+  const handleRef = useForkRef(
+    React.isValidElement(children) ? getElementRef(children) : null,
+    forwardedRef,
+  );
```

Comment:
```suggestion
  const handleRef = useForkRef(getElementRef(children), forwardedRef);
```

We can likely simplify, as the `getElementRef` handles this already.

Reply:
I refactored to handle this case and renamed the util accordingly 😊